app/scenes/Search/Search.js

```diff
@@ -33,349 +28,352 @@ import DateFilter from "./components/DateFilter";
-        this.isLoading = false;
+      if (lastQuery === query && isEqual(params, lastParams)) {
+        setIsLoading(false);
         return;
       }
 
-      this.isLoading = true;
-      this.lastQuery = this.query;
-      this.lastParams = params;
+      setIsLoading(true);
+      lastQuery.current = query;
+      lastParams.current = params;
 
       try {
-        const results = await this.props.documents.search(this.query, params);
+        setResults(await documents.search(query, params));
 
-        this.pinToTop = true;
+        setPinToTop(true);
 
         if (results.length === 0 || results.length < DEFAULT_PAGINATION_LIMIT) {
-          this.allowLoadMore = false;
+          setAllowLoadMore(false);
         } else {
-          this.offset += DEFAULT_PAGINATION_LIMIT;
+          setOffset((preOffset) => preOffset + DEFAULT_PAGINATION_LIMIT);
         }
       } catch (err) {
-        this.lastQuery = "";
+        lastParams.current = "";
         throw err;
       } finally {
-        this.isLoading = false;
+        setIsLoading(false);
       }
     } else {
-      this.pinToTop = false;
-      this.lastQuery = this.query;
+      setPinToTop(false);
+      lastQuery.current = query;
     }
-  };
-
-  fetchResultsDebounced = debounce(this.fetchResults, 500, {
+  }, [
+    collectionId,
+    dateFilter,
+    userId,
+    documents,
+    includeArchived,
+    offset,
+    query,
+    results.length,
+  ]);
+
+  const fetchResultsDebounced = debounce(fetchResults, 500, {
     leading: false,
     trailing: true,
   });
 
-  updateLocation = (query: string) => {
-    this.props.history.replace({
-      pathname: searchUrl(query),
-      search: this.props.location.search,
-    });
-  };
+  const handleQueryChange = React.useCallback(() => {
+    setParams(new URLSearchParams(location.search));
+    setOffset(0);
+    setAllowLoadMore(true);
+
+    // To prevent "no results" showing before debounce kicks in
+    setIsLoading(true);
+
+    fetchResultsDebounced();
+  }, [location.search, fetchResultsDebounced]);
+
+  const handleTermChange = React.useCallback(() => {
+    const query = decodeURIComponentSafe(match.params.term || "");
+    setQuery(query ? query : "");
+    setOffset(0);
+    setAllowLoadMore(true);
+
+    // To prevent "no results" showing before debounce kicks in
+    setIsLoading(!!query);
 
-  setFirstDocumentRef = (ref: any) => {
-    this.firstDocument = ref;
+    fetchResultsDebounced();
+  }, [match.params.term, fetchResultsDebounced]);
+
+  React.useEffect(() => {
+    if (locationSearchRef.current !== location.search) {
+      handleQueryChange();
+      locationSearchRef.current = location.search;
+    }
+  }, [handleQueryChange, location.search, locationSearchRef]);
+
+  React.useEffect(() => {
+    if (matchParamsTermRef.current !== match.params.term) {
+      handleTermChange();
+      matchParamsTermRef.current = match.params.term;
+    }
+  }, [handleTermChange, match.params.term]);
```

Comment:
These refs shouldn't be needed, with hooks you can just make eg `location.search` one of the deps and the hook will only be called when it changes – so there's no need to store or compare against the previous value like with `componentDidUpdate`